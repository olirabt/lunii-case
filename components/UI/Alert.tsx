import React from "react";
import cn from "classnames";

interface AlertProps {
	type: "success" | "warning";
	children: React.ReactNode;
}

const Alert: React.FC<AlertProps> = ({ type, children }) => {

	const alertClass = {
		"bg-validate500 text-validate400": type === "success",
		"bg-alert400 text-alert500": type === "warning",
	};

	return (
		<div className={cn("text-left", "w-full", "px-5", "py-3", "mx-auto", "rounded-md", alertClass)}>
			{children}
		</div>
	);
};

export default Alert;