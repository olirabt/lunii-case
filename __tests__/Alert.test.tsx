import { render } from '@testing-library/react'
import Alert from "@/components/UI/Alert";

describe("Alert", () => {
  it("renders the success message correctly", () => {
    const { container } = render(<Alert type="success"><p>Success message</p></Alert>)
    expect(container).toMatchSnapshot();
  });

  it("renders the warning message correctly", () => {
    const { container } = render(<Alert type="warning"><p>L&apos;url saisie est invalide</p></Alert>)
    expect(container).toMatchSnapshot();
  });
});
