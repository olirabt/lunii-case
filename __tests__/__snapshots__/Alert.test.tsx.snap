// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert renders the success message correctly 1`] = `
<div>
  <div
    class="text-left w-full px-5 py-3 mx-auto rounded-md bg-validate500 text-validate400"
  >
    <p>
      Success message
    </p>
  </div>
</div>
`;

exports[`Alert renders the warning message correctly 1`] = `
<div>
  <div
    class="text-left w-full px-5 py-3 mx-auto rounded-md bg-alert400 text-alert500"
  >
    <p>
      L'url saisie est invalide
    </p>
  </div>
</div>
`;
